File: baekjoon/브루트포스 알고리즘/[10448]유레카이론.py
T = [n*(n+1)//2 for n in range(1, 46)]
li = [0] * 1001

for i in T:
    for j in T:
        for k in T:
            if i+j+k <= 1000:
                li[i+j+k] = 1
                
num = int(input())

for i in range(num):
    print(li[int(input())])    

File: baekjoon/브루트포스 알고리즘/[10448]유레카이론.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<int> v;

int find(int num) {
    for (int i = 0; i < v.size(); i++) {
        for (int j = 0; j < v.size(); j++) {
            for (int k = 0; k < v.size(); k++) {
                if (v[i] + v[j] + v[k] == num) {
                    
                    return 1;
                }
            }
        }
    }
    return 0;
}

int main() {
    int t, num;    

    cin >> t;
    for (int i = 1; i < 47; i++) {  
        int n;
        n = i * (i + 1) / 2;

        v.push_back(n);
    }

    for (int i = 0; i < t; i++) {
        cin >> num;
        cout << find(num) << "\n";
    }

    return 0;
}